examples/platforms/posix/radio.c

```diff
@@ -715,7 +714,13 @@ void radioSendAck(void)
     sAckFrame.mLength    = IEEE802154_ACK_LENGTH;
     sAckMessage.mPsdu[0] = IEEE802154_FRAME_TYPE_ACK;
 
-    if (isDataRequestAndHasFramePending(&sReceiveFrame))
+    if (
+#if OPENTHREAD_THREAD_VERSION >= OPENTHREAD_THREAD_VERSION_1_2
+        (otMacFrameIsDataRequest(&sReceiveFrame) || otMacFrameIsDataRequest(&sReceiveFrame))
```

Comment:
Shouldn't this be like this?:

```suggestion
        (otMacFrameIsDataRequest(&sReceiveFrame) || otMacFrameIsData(&sReceiveFrame))
```